webRequestTranslator/source/webRequestTranslator.cc

```diff
@@ -0,0 +1,189 @@
+#include "webRequestTranslator.h"
+
+namespace webRequestTranslator {
+    /**
+     * Converts the RDFS namenode create response into the appropriate webRDFS response.
+     */
+    std::string getNamenodeCreateResponse(hadoop::hdfs::DatanodeInfoProto &dataProto, std::string requestLink) {
+      std::string res = std::string("HTTP/1.1 307 TEMPORARY_REDIRECT\n");
+
+      std::string delimiter = "/webhfs/v1/";
+      std::string restOfRequest = requestLink.substr(requestLink.find(delimiter) + delimiter.length(), requestLink.length());
+
+      hadoop::hdfs::DatanodeIDProto id = dataProto.id();
+      res += "Location: http://";
+      res += id.hostname();
+      res += ":";
+      res += std::to_string(id.infoport());
+
+      res += delimiter;
+      res += restOfRequest;
+
+      res += "\nContent-Length: 0";
+      return res;
+    }
+
+    /**
+     * Converts the RDFS datanode create response into the appropriate webRDFS response.
+     */
+    std::string getDatanodeCreateResponse(std::string location, std::string contentOfFile) {
+      std::string res = std::string("HTTP/1.1 200 OK\nLocation: ");
+
+      res += location;
+      res += "\n";
+      res += "Content-Length: ";
+      res += std::to_string(contentOfFile.length());
+      res += "\n\n";
+
+      return res;
+    }
+  /**
+   * Converts the RDFS namenode read response into the appropriate webRDFS response.
+   */
+  std::string getNamenodeReadResponse(hadoop::hdfs::DatanodeInfoProto &dataProto, std::string requestLink) {
+    std::string res = std::string("HTTP/1.1 307 TEMPORARY_REDIRECT\n");
```

Comment:
`s/_/ /g`

to conform to HTTP standard, [reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307)